src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -45,39 +51,91 @@ public Task<ActionResult> GetDump(int? pid, [FromQuery]DumpType type = DumpType.
+                    dumpFileName = $"{process.ProcessName}.{pidValue}.dmp";
                 }
                 else
                 {
-                    dumpFileName = $"core_{result.Pid}";
+                    dumpFileName = $"core_{pidValue}";
                 }
 
-                return CreateFileStreamResult(result, dumpFileName);
+                //Compression is done automatically by the response
+                //Chunking is done because the result has no content-length
+                return File(result, "application/octet-stream", Invariant(dumpFileName));
             });
         }
 
         [HttpGet("cpuprofile/{pid?}")]
-        public Task<ActionResult> CpuProfile(int? pid, [FromQuery]int duration = 30)
+        public Task<ActionResult> CpuProfile(int? pid, [FromQuery]int durationSeconds = 30)
+        {
+            return InvokeService(async () =>
+            {
+                ValidateDuration(durationSeconds);
+                int pidValue = _diagnosticServices.ResolveProcess(pid);
+                IStreamWithCleanup result = await _diagnosticServices.StartCpuTrace(pidValue, durationSeconds, this.HttpContext.RequestAborted);
+                return new StreamWithCleanupResult(result, "application/octet-stream", Invariant($"{Guid.NewGuid()}.nettrace"));
+            });
+        }
+
+        [HttpGet("trace/{pid?}")]
+        public Task<ActionResult> Trace(int? pid, [FromQuery]int durationSeconds = 30)
         {
             return InvokeService(async () =>
             {
-                OperationResult<Stream> result = await _diagnosticServices.StartCpuTrace(pid, duration, this.HttpContext.RequestAborted);
-                return CreateFileStreamResult(result, $"{Guid.NewGuid()}.nettrace");
+                ValidateDuration(durationSeconds);
+                int pidValue = _diagnosticServices.ResolveProcess(pid);
+                IStreamWithCleanup result = await _diagnosticServices.StartTrace(pidValue, durationSeconds, this.HttpContext.RequestAborted);
+                return new StreamWithCleanupResult(result, "application/octet-stream", Invariant($"{Guid.NewGuid()}.nettrace"));
+            });
+        }
+
+        [HttpGet("logs/{pid?}")]
+        public ActionResult Logs(int? pid, [FromQuery] int durationSeconds = 30)
+        {
+            return InvokeService(() =>
+            {
+                ValidateDuration(durationSeconds);
+                int pidValue = _diagnosticServices.ResolveProcess(pid);
+                return new OutputStreamResult(async (outputStream, token) =>
+                {
+                    await _diagnosticServices.StartLogs(outputStream, pidValue, durationSeconds, token);
+                }, "application/x-ndjson", Invariant($"{Guid.NewGuid()}.txt"));
             });
         }
 
-        private FileStreamResult CreateFileStreamResult(OperationResult<Stream> result, FormattableString downloadName)
+        private static void ValidateDuration(int durationSeconds)
```

Comment:
Instead of doing custom validation like this, you can apply the Range attribute to the durationSeconds parameters, which will enforce the allowable range via model validation. For example:

`public ActionResult Logs(int? pid, [FromQuery, Range(0, Int32.MaxValue)] int durationSeconds = 30)`